packages/hoppscotch-backend/src/shortcode/shortcode.service.ts

```diff
@@ -263,4 +283,55 @@ export class ShortcodeService implements UserDataHandler, OnModuleInit {
       return E.left(SHORTCODE_NOT_FOUND);
     }
   }
+
+  /**
+   * Fetch all created ShortCodes
+   *
+   * @param args Pagination arguments
+   * @param userEmail User email
+   * @returns ShortcodeWithUserEmail
+   */
+  async fetchAllShortcodes(
+    args: PaginationArgs,
+    userEmail: string | null = null,
+  ) {
+    const shortCodes = await this.prisma.shortcode.findMany({
+      where: userEmail
+        ? {
+            User: {
+              email: userEmail,
+            },
+          }
+        : undefined,
+      orderBy: {
+        createdOn: 'desc',
+      },
+      skip: args.cursor ? 1 : 0,
+      take: args.take,
+      cursor: args.cursor ? { id: args.cursor } : undefined,
+      include: {
+        User: true,
+      },
+    });
+
+    const fetchedShortCodes: ShortcodeWithUserEmail[] = shortCodes.map(
+      (code) => {
+        return <ShortcodeWithUserEmail>{
+          id: code.id,
+          request: JSON.stringify(code.request),
+          properties:
+            code.embedProperties != null
+              ? JSON.stringify(code.embedProperties)
+              : null,
+          createdOn: code.createdOn,
+          creator: {
+            uid: code.User.uid,
+            email: code.User.email,
+          },
```

Comment:
Can we create a `@ResolveField` for this instead of joining? What do you think?

Reply:
Nope, I don't think it's necessary. By creating this as a field resolver it will be available to all the created shortcodes, we only want this property to be available to that one specific query in the admin module.